src/main/java/com/rickmorty/Services/CharacterService.java

```diff
@@ -44,21 +51,29 @@ public ApiResponseDto<CharacterDto> findAllCharacters(Integer page) {
             ApiResponseDto<CharacterDto> apiResponseDto = objectMapper.readValue(response.body(),
                     new TypeReference<ApiResponseDto<CharacterDto>>() {
                     });
+
+            if (response.statusCode() != 200) throw new PageNotFoundException();
```

Comment:
mudar para se o status for 404 por que caso o pior ocorra e api externa caia iria retornar 404 pro usuário apesar de não ser um erro dele